### PR TITLE
Textarea kit created

### PR DESCRIPTION
**Pull request by christinaatai**
Screen Shot 2019-12-02 at 11 45 14 AM

### CODE REVIEW

File: app/pb_kits/playbook/pb_body/_body_mixins.scss [hunk 1]
```diff
@@ -21,8 +21,11 @@ $pb_body_status: (
   line-height: $lh_tight;
   color: $color;
   font-size: $text_default;
+  font-weight: $regular;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_body/_body_mixins.scss]: I thought you were adding font-family here

File: app/pb_kits/playbook/pb_textarea/_textarea.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    class: object.classname) do %>
+  <% if object.label.present? %>
+    <%= pb_rails("caption", props: {text: object.label, dark: object.dark}) %>
+  <% end %>
+  <% if object.children %>
+    <%= capture(&object.children) %>
+  <% else %>
+    <%= text_area(
+        nil, 
+        nil,
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.html.erb]: @thestephenmarshall Thoughts?

File: app/pb_kits/playbook/pb_textarea/_textarea.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  <% if object.label.present? %>
+    <%= pb_rails("caption", props: {text: object.label, dark: object.dark}) %>
+  <% end %>
+  <% if object.children %>
+    <%= capture(&object.children) %>
+  <% else %>
+    <%= text_area(
+        nil, 
+        nil, 
+        :class => "#{object.classname} form-control",
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.html.erb]: Why are you adding this class? This is an old bootstrap pattern.

File: app/pb_kits/playbook/pb_textarea/_textarea.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+const Textarea = ( props: TextareaProps) => {
+
+  const {
+    className,
+    label,
+    placeholder,
+    value,
+    dark=false,
+    rows = 4,
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.jsx]: correct the spacing

File: app/pb_kits/playbook/pb_textarea/_textarea.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  } = props
+
+  const textarea_input = `${textareaCSS(props)} form-control`
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.jsx]: remove `form-control` class

File: app/pb_kits/playbook/pb_textarea/_textarea.jsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+type TextareaProps = {
+  className?: String,
+  data?: String,
+  id?: String,
+  label?: String,
+  placeholder?: String,
+  value?: String,
+  rows?: Number,
+  dark?: Boolean,
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.jsx]: Missing name

File: app/pb_kits/playbook/pb_textarea/_textarea.html.erb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    id: object.id,
+    data: object.data,
+    class: object.classname) do %>
+  <% if object.label.present? %>
+    <%= pb_rails("caption", props: {text: object.label, dark: object.dark}) %>
+  <% end %>
+  <% if object.children %>
+    <%= capture(&object.children) %>
+  <% else %>
+    <%= text_area(
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.html.erb]: Missing name

File: app/pb_kits/playbook/pb_textarea/_textarea.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  
+  return (
+    <div className={classnames(textareaCSS(props), className)}>
+      <Caption text={label} dark={dark}/>
+      <textarea className={textarea_input}
+        placeholder={placeholder}
+        rows={rows}>
+        {value}
+      </textarea>
+    </div>
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.jsx]: This does not take a block

File: app/pb_kits/playbook/pb_textarea/_textarea_mixin.scss [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+@mixin pb_textarea($border_color, $background_color, $text_color){
+    @include pb_title_4;
+    border: 1px solid $border_color;
+    border-radius: $border_rad_light;
+    display: block;
+    width: 100%;
+    color: $text_color;
+    background: $background_color;
+    padding: $space_xs $space_sm;
+    transition: 0.2s all;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea_mixin.scss]: @markiearnold should this use one of the animation variables?

**Comment by jasoncypret** [comment]
Can you add a screenshot + add an example where you are adding your own input. Like toggle "Custom checkbox input" https://playbook.powerapp.cloud/kits/toggle

File: app/pb_kits/playbook/pb_textarea/textarea.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Playbook
+  module PbTextarea
+    class Textarea
+      include Playbook::Props
+
+      partial "pb_textarea/textarea"
+
+      prop :label
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/pb_textarea/textarea.rb]:
```suggestion
      prop :object
      prop : name
      prop :label
```

File: app/pb_kits/playbook/pb_textarea/_textarea.html.erb (hunk 2 above)
Comment (thestephenmarshall) [reply app/pb_kits/playbook/pb_textarea/_textarea.html.erb]: These need to be props [as outlined here](https://apidock.com/rails/ActionView/Helpers/FormHelper/text_area) `text_area(object_name, method...` are the arguments needed to properly build the textarea attributes `name` and `id` which are required in Rails forms. I haven't tried with strings before, but they should work

File: app/pb_kits/playbook/pb_textarea/textarea.rb (hunk 10 above)
Comment (thestephenmarshall) [reply app/pb_kits/playbook/pb_textarea/textarea.rb]: These props need to be added as mentioned in `text_area.html.erb`

File: spec/pb_kits/playbook/kits/textarea_spec.rb [hunk 11]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require_relative "../../../../app/pb_kits/playbook/pb_textarea/textarea"
+
+RSpec.describe Playbook::PbTextarea::Textarea do
+  subject { Playbook::PbTextarea::Textarea }
+
+  it { is_expected.to define_partial }
+
+  it { is_expected.to define_string_prop(:label) }
```
Comment (thestephenmarshall) [inline spec/pb_kits/playbook/kits/textarea_spec.rb]: You will have to add some expectations to cover your new props here.

File: app/pb_kits/playbook/pb_textarea/textarea.rb (hunk 10 above)
Comment (christinaatai) [reply app/pb_kits/playbook/pb_textarea/textarea.rb]: @thestephenmarshall Why is object a prop?

File: app/pb_kits/playbook/pb_textarea/_textarea_mixin.scss (hunk 9 above)
Comment (christinaatai) [reply app/pb_kits/playbook/pb_textarea/_textarea_mixin.scss]: I changed transition to $easeIn for now, as transitions are going to be further explored.

File: app/pb_kits/playbook/pb_textarea/_textarea.jsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    name,
+  } = props
+
+  const textarea_input = `${textareaCSS(props)}`
+  
+  return (
+    <div className={classnames(textareaCSS(props), className)}>
+      <Caption text={label} dark={dark}/>
+      <textarea className={textarea_input} name={name} placeholder={placeholder} rows={rows}>
+        {value || children}
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/_textarea.jsx]: This needs to wrap around the textarea...So if `children` are passed we display those `{children }` else display your textarea. See some example if logic: https://github.com/powerhome/playbook/blob/e5f825652835ae3da4fe28d4e13e0acb705b1bab/app/pb_kits/playbook/pb_dashboard_value/_dashboard_value.jsx#L49

File: app/pb_kits/playbook/pb_textarea/docs/_textarea_custom.jsx [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import React from "react"
+import {Textarea} from "../../"
+
+function TextareaCustom() {
+  return (
+    <div>
+        <Textarea label="Label">
+            Default text
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/docs/_textarea_custom.jsx]: This should be passing a textarea

File: app/pb_kits/playbook/pb_textarea/docs/_textarea_custom.jsx [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import React from "react"
+import {Textarea} from "../../"
+
+function TextareaCustom() {
+  return (
+    <div>
+        <Textarea label="Label" />
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_textarea/docs/_textarea_custom.jsx]: Can you make this a block example:

```
<Textarea label="Label">
    <textarea class="my_custom_class" name="custom_textarea" rows=4></textarea>
</Textarea>
```

**Pull request merged**